DuosForSums.java

```diff
@@ -0,0 +1,73 @@
+import java.util.Arrays;
+
+public class DuosForSums {
+
+    public static void main(String[] args) {
+        int[] initArr = {1, 3, 5, -2, 6, 0, -3, -1, 7, 2};
+        int sumNum = 5;
+        sort(initArr);
+        group(initArr, sumNum);
+    }
+
+
+    private static void swap(int[] array, int ind1, int ind2) {
+        int tmp = array[ind1];
+        array[ind1] = array[ind2];
+        array[ind2] = tmp;
+    }
+
+    private static void sort(int[] array) {

```

Comment:
It is not necessary to implement custom sorting, use Collections.sort(..) where complexity is nlog(n)

Reply:
Meanwhile sorting here is good optimization!